src/client/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -19,8 +19,10 @@ import { IDebuggingCellMap, IJupyterSession } from '../../datascience/types';
 import { Kernel, KernelMessage } from '@jupyterlab/services';
 import { ICommandManager } from '../../common/application/types';
 import { traceError } from '../../common/logger';
+import { IFileSystem } from '../../common/platform/types';
 
 const debugRequest = (message: DebugProtocol.Request): KernelMessage.IDebugRequestMsg => {
+    const sessionId = message.arguments?.__sessionId ? message.arguments.__sessionId : randomBytes(8).toString('hex');
```

Comment:
Hang on a second, shouldn't this be the Jupyter session ID?

Reply:
this will be sent to jupyter, and if the message came from jupyter the casting should put it there. But I'll test and make sure.

Reply:
so, sessionId only comes in the attach message. I think I should save it and send the same one but I'm not sure. I'll investigate.